src/Razor/Microsoft.AspNetCore.Razor.Language/src/Legacy/CSharpCodeParser.cs

```diff
@@ -1459,6 +1470,52 @@ private void ParseExtensibleDirective(in SyntaxListBuilder<RazorSyntaxNode> buil
                                     return;
                                 }
 
+                                break;
+                            case DirectiveTokenKind.GenericTypeConstraint:
+                                if (At(SyntaxKind.Keyword) &&
+                                    string.Equals(CurrentToken.Content, CSharpLanguageCharacteristics.GetKeyword(CSharpKeyword.Where), StringComparison.Ordinal))
+                                {
+                                    // Consume the 'where' keyword plus any aditional whitespace
+                                    AcceptAndMoveNext();
+                                    AcceptWhile(SyntaxKind.Whitespace);
+                                    // Check that the type name matches the type name before the where clause.
+                                    // Find a better way to do this
+                                    if (!string.Equals(CurrentToken.Content, lastSeenMemberIdentifier, StringComparison.Ordinal))
+                                    {
+                                        // @typeparam TKey where TValue : ...
+                                        // The type parameter in the generic type constraint 'TValue' does not match the type parameter 'TKey' defined in the directive '@typeparam'.
+                                        Context.ErrorSink.OnError(
+                                            RazorDiagnosticFactory.CreateParsing_GenericTypeParameterIdentifierMismatch(
+                                                new SourceSpan(CurrentStart, CurrentToken.Content.Length), descriptor.Directive, CurrentToken.Content, lastSeenMemberIdentifier));
+                                        builder.Add(BuildDirective());
+                                        return;
+                                    }
+                                    else
```

Comment:
Do we need this here if we're consuming everything until the end of line later?

Reply:
Good call, I was following my mental parsing model of, consume where, consume constraint identifier, consume the rest

Reply:
done